### PR TITLE
New Zealand: optimizations

### PR DESCRIPTION
**Pull request by KJhellico**

### CODE REVIEW

File: holidays/countries/new_zealand.py [hunk 1]
```diff
@@ -78,9 +76,14 @@ class NewZealand(HolidayBase):
         "WTL",  # Correct code is WTC
     ]
 
-    def _populate(self, year):
-        super()._populate(year)
+    @staticmethod
```
Comment (arkid15r) [inline holidays/countries/new_zealand.py:79]: It seems like a good candidate for DRY optimization (saw similar code in canada.py). No action required, just saying.

**Review by arkid15r — COMMENTED** [review]
Looks good 👍

File: holidays/countries/new_zealand.py (hunk 1 above)
Comment (KJhellico) [reply holidays/countries/new_zealand.py:79]: Sorry, I didn't quite understand what it was about :(

Comment (arkid15r) [reply holidays/countries/new_zealand.py:79]: We're going to have exactly the same code in 2 different places. Could be extracted and reused.

Comment (KJhellico) [reply holidays/countries/new_zealand.py:79]: You mean `_get_nearest_monday`, I got it! :) Should I move it to utils.py?

**Pull request merged**